sklearn/ensemble/_hist_gradient_boosting/tests/test_gradient_boosting.py

```diff
@@ -1178,3 +1178,31 @@ def test_class_weights():
         clf_balanced.decision_function(X_imb),
         clf_sample_weight.decision_function(X_imb),
     )
+
+
+def test_unknown_category_that_are_negative():
+    """Check that unknown categories that are negative does not error.
+
+    Non-regression test for #24274.
+    """
+
+    rng = np.random.RandomState(0)
+    n_samples = 1000
+    X = np.c_[rng.rand(n_samples), rng.randint(4, size=n_samples)]
+    y = np.zeros(shape=n_samples)
+    y[X[:, 1] % 2 == 0] = 1
+
+    hist = HistGradientBoostingRegressor(
+        random_state=0,
+        categorical_features=[False, True],
+        max_iter=10,
+    ).fit(X, y)
+
+    # Negative values from the second column are treated as unknown
+    X_test_neg = np.asarray([[1, -2], [3, -4]])
+    y_pred_neg_one = hist.predict(X_test_neg)
```

Comment:
The main goal is that is does not error. But can we assert something about the predicted values themselves?

Reply:
A few lines down, it asserts that the negative categories are the same as the result for `nan`.

Reply:
I updated the PR with [`e300ee3` (#24283)](https://github.com/scikit-learn/scikit-learn/pull/24283/commits/e300ee34dfd259bbc24e033704c004547b63302b) to clarify the test.